source/opensslcrypto/opensslcrypto_ecc.c

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <aws/cal/ecc.h>
+
+#include <aws/cal/cal.h>
+
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+#include <openssl/obj_mac.h>
+
+struct libcrypto_ecc_key {
+    struct aws_ecc_key_pair key_pair;
+    EC_KEY *ec_key;
```

Comment:
Q: OpenSSL states that their long term support would be for the more general `EVP_PKEY*`  apis and not the low level `EC_KEY*`  apis ones. The EC_KEY apis are opaque in the newer long term supported version. Given this, do we want to use the more generic `EVP_PKEY *`?

Reply:
This question is not at all related to this PR. As for the question itself, it is very likely that we will need to move to the ```EVP_PKEY``` enums, and @JonathanHenson would have more of an opinion on when we should do that. We should probably move this conversation to an issue.

Reply:
I evaluated the API and due to the operations we're using we still had to fallback to the low-level interfaces. The implementation is completely hidden, so worst case we'll have to revisit this in the future, but it shouldn't break anything architecturally to do so.